### PR TITLE
update API to use boost::signals2 (#267)

### PR DESCRIPTION
**Pull request by dirk-thomas**

### CODE REVIEW

**Comment by dirk-thomas** [comment]
@ros-pull-request-builder Please run CI on this pull request...

**Comment by dirk-thomas** [comment]
This is breaking API between _Hydro_ and _indigo_ (and I can't imagine a way to avoid it).

@tfoote @wjwwood Please review and vote on API breaking change for _Indigo_.

**Comment by tfoote** [comment]
+1 Boost is changing we'll need to do it anyway. If using the typedef it should keep working.

File: utilities/message_filters/include/message_filters/sync_policies/exact_time.h [hunk 1]
```diff
@@ -45,15 +45,7 @@
...
-#ifndef BOOST_SIGNALS_NO_DEPRECATION_WARNING
-#define BOOST_SIGNALS_NO_DEPRECATION_WARNING
-#define ros_BOOST_SIGNALS_NO_DEPRECATION_WARNING
-#endif
-#include <boost/signals.hpp>
-#ifdef ros_BOOST_SIGNALS_NO_DEPRECATION_WARNING
-#undef ros_BOOST_SIGNALS_NO_DEPRECATION_WARNING
-#undef BOOST_SIGNALS_NO_DEPRECATION_WARNING
-#endif
+#include <boost/signals2.hpp>
```
Comment (wjwwood) [inline utilities/message_filters/include/message_filters/sync_policies/exact_time.h]: Does this header (and similar headers here) even need this include?

**Comment by wjwwood** [comment]
+1 from me, I think this change needs to go in, we just need to update the Migration guide. My guess is that very few people actually use this API directly.

**Comment by dirk-thomas** [comment]
I removed the unnecessary includes from most of the message_filters headers.

**Comment by wjwwood** [comment]
+1

**Comment by dirk-thomas** [comment]
@ros-pull-request-builder

**Pull request merged**